x-pack/legacy/plugins/apm/public/components/shared/charts/TransactionCharts/TransactionLineChart/index.tsx

```diff
@@ -26,82 +21,37 @@ interface Props {
     data: Array<Coordinate | RectCoordinate>;
     type: string;
   }>;
-  stacked?: boolean;
   truncateLegends?: boolean;
-  tickFormatY: (y: number | null) => React.ReactNode;
+  tickFormatY: (y: number | null | undefined) => React.ReactNode;
   formatTooltipValue: (c: Coordinate) => React.ReactNode;
   yMax?: string | number;
   height?: number;
+  stacked?: boolean;
 }
 
 const TransactionLineChart: React.FC<Props> = (props: Props) => {
   const {
     series,
     tickFormatY,
     formatTooltipValue,
-    stacked = false,
     yMax = 'max',
     height,
-    truncateLegends
+    truncateLegends,
+    stacked = false
   } = props;
 
-  const flattenedCoordinates = flatten(series.map(serie => serie.data));
-
-  const start = d3.min(flattenedCoordinates, d => d.x);
-  const end = d3.max(flattenedCoordinates, d => d.x);
-
-  const noHits = series.every(
-    serie =>
-      serie.data.filter(value => 'y' in value && value.y !== null).length === 0
-  );
-  const { time, setTime } = useChartsTime();
-
-  const hoverXHandlers = useMemo(() => {
-    return {
-      onHover: (hoverX: number) => {
-        setTime(hoverX);
-      },
-      onMouseLeave: () => {
-        setTime(null);
-      },
-      onSelectionEnd: (range: { start: number; end: number }) => {
-        setTime(null);
-
-        const currentSearch = toQuery(history.location.search);
-        const nextSearch = {
-          rangeFrom: new Date(range.start).toISOString(),
-          rangeTo: new Date(range.end).toISOString()
-        };
-
-        history.push({
-          ...history.location,
-          search: fromQuery({
-            ...currentSearch,
-            ...nextSearch
-          })
-        });
-      },
-      hoverX: time
-    };
-  }, [time, setTime]);
-
-  if (!start || !end) {
-    return null;
-  }
+  const { hoverXHandlers } = useChartsSync();
 
   return (
     <CustomPlot
-      noHits={noHits}
-      series={noHits ? getEmptySeries(start, end) : series}
-      start={new Date(start).getTime()}
-      end={new Date(end).getTime()}
+      series={series}
       {...hoverXHandlers}
       tickFormatY={tickFormatY}
       formatTooltipValue={formatTooltipValue}
-      stacked={stacked}
       yMax={yMax}
       height={height}
       truncateLegends={truncateLegends}
+      {...(stacked ? { stackBy: 'y' } : {})}
```

Comment:
What about:
```js
stackBy: {stacked ? 'y' : undefined}
```

Reply:
or perhaps just pass in `stacked` and handle it in plotUtils but with `undefined` as the fallback value.